examples/md-flexible/Thermostat.h

```diff
@@ -137,48 +95,32 @@ void Thermostat<AutoPasTemplate, ParticlePropertiesLibraryTemplate>::addBrownian
   }
 }
 
-template <class AutoPasTemplate, class ParticlePropertiesLibraryTemplate>
-void Thermostat<AutoPasTemplate, ParticlePropertiesLibraryTemplate>::apply(AutoPasTemplate &autopas) {
-  ThermostatFloatType temp = calcTemperature(autopas);
-  ThermostatFloatType scaling;
-  if (_tInit == _tTarget) {
-    scaling = std::sqrt(_tTarget / temp);
-  } else {
-    if (std::abs(_tInit + _deltaTemp) > std::abs(_tTarget)) {
-      _tInit = _tTarget;
-    } else {
-      _tInit = _tInit + _deltaTemp;
-    }
-    scaling = std::sqrt(_tInit / temp);
+/**
+ * Scales velocity of particles towards a gived temperature.
+ * @tparam AutoPasTemplate Type of AutoPas Object (no pointer)
+ * @tparam ParticlePropertiesLibraryTemplate Type of ParticlePropertiesLibrary Object (no pointer)
+ * @param autopas
+ * @param particlePropertiesLibrary
+ * @param targetTemperature
+ * @param deltaTemperature Maximum temperature change.
+ */
+template<class AutoPasTemplate, class ParticlePropertiesLibraryTemplate>
+void apply(AutoPasTemplate &autopas,
```

Comment:
here, you need to more or less apply the thermostat for each component separately.
=> separate temperatures and scaling factors for each component!